core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -194,12 +194,32 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv, numUsableCores: Int) exte
     endpoints.containsKey(name)
   }
 
-  /** Thread pool used for dispatching messages. */
-  private val threadpool: ThreadPoolExecutor = {
+  def getNumOfThreads(conf: SparkConf): Int = {
     val availableCores =
       if (numUsableCores > 0) numUsableCores else Runtime.getRuntime.availableProcessors()
-    val numThreads = nettyEnv.conf.get(RPC_NETTY_DISPATCHER_NUM_THREADS)
+    // module configuration
+    val modNumThreads = nettyEnv.conf.get(RPC_NETTY_DISPATCHER_NUM_THREADS)
       .getOrElse(math.max(2, availableCores))
+    // try to get specific threads configurations of driver and executor
+    // override module configurations if specified
+    val executorId = conf.get("spark.executor.id", "")
+    // neither driver nor executor if executor id is not set
```

Comment:
comment just repeats the code.

Reply:
you mean I should wrap the logic in one method for multiple invocation?

Reply:
let me remove duplicate comments